Access SAML group is causing panic

1) my bad: looks like I'm actually "just" still have panic crashes ...
2) here's the output, in case it helps (it's not the full output - just the relevant parts; hopefully, that should be enough; this is because the less I put here, the less I have to sanitize and the less I might leak important data)

```
2021-02-11T13:44:24.116+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: -----------------------------------------------------
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 2021/02/11 13:44:24 [DEBUG] Cloudflare API Response Details:
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: ---[ RESPONSE ]--------------------------------------
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: HTTP/1.1 200 OK
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Connection: close
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Transfer-Encoding: chunked
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Cf-Cache-Status: DYNAMIC
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Cf-Ray: 61faba535bde17c8-MEL
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Cf-Request-Id: 08MASKED0001
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Content-Security-Policy: frame-ancestors 'none'; default-src https: 'unsafe-inline'
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Content-Type: application/json; charset=UTF-8
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Date: Thu, 11 Feb 2021 02:44:24 GMT
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Expect-Ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Server: cloudflare
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Set-Cookie: __cfduid=d7dd0eb9MASKED13011464; expires=Sat, 13-Mar-21 02:44:24 GMT; path=/; domain=.api.cloudflare.com; HttpOnly; SameSite=Lax; Secure
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Set-Cookie: __cflb=0H28vMASKEDq2zy; SameSite=Lax; path=/; expires=Thu, 11-Feb-21 05:14:25 GMT; HttpOnly
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Set-Cookie: __cfruid=fcd2c00MASKED464; path=/; domain=.api.cloudflare.com; HttpOnly; Secure; SameSite=None
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: Strict-Transport-Security: max-age=31536000; includeSubDomains
form-provider-cloudflare_v2.18.0: Vary: Accept-Encoding
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: X-Content-Type-Options: nosniff
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: X-Envoy-Upstream-Service-Time: 6
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: X-Frame-Options: DENY
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: X-Xss-Protection: 1; mode=block
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 1eb
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: {
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:   "result": {
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "id": "MASKED",
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "name": "ba2_role",
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "uid": "MASKED",
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "include": [
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:       {
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:         "saml": {
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:           "attribute_name": "memberOf",
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:           "attribute_value": "ba2_role"
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:         }
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:       }
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     ],
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "require": [],
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "exclude": [],
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "created_at": "2021-02-11T02:44:24Z",
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:     "updated_at": "2021-02-11T02:44:24Z"
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:   },
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:   "success": true,
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:   "errors": [],
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:   "messages": []
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: }
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 0
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 
2021-02-11T13:44:24.440+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: -----------------------------------------------------

Warning: Resource targeting is in effect

You are creating a plan with the -target option, which means that the result
of this plan may not represent all of the changes requested by the current
configuration.

The -target option is not for routine use, and is provided only for
exceptional situations such as recovering from errors or mistakes, or when
Terraform specifically suggests to use it as part of an error message.


Warning: Applied changes may be incomplete

The plan was created with the -target option in effect, so some changes
requested in the configuration may have been ignored and the output values may
not be fully updated. Run the following command to verify that no other
changes are pending:
    terraform plan

Note that the -target option is not suitable for routine use, and is provided
only for exceptional situations such as recovering from errors or mistakes, or
when Terraform specifically suggests to use it as part of an error message.


Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not string
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: 
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: goroutine 81 [running]:
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.TransformAccessGroupForSchema(0xc0000c2840, 0x1, 0x4, 0x110de80, 0xc00063a880, 0x0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_access_group.go:611 +0x524d
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareAccessGroupRead(0xc000572af0, 0x132fa00, 0xc00017e780, 0x0, 0x0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_access_group.go:261 +0x87a
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareAccessGroupCreate(0xc000572af0, 0x132fa00, 0xc00017e780, 0x2, 0x1c385a0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_access_group.go:294 +0x5ba
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000172990, 0xc000109b30, 0xc000636b40, 0x132fa00, 0xc00017e780, 0x1198301, 0xc000734898, 0xc00060fd10)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/hashicorp/terraform-plugin-sdk@v1.16.0/helper/schema/resource.go:310 +0x375
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00021a300, 0xc0006efa38, 0xc000109b30, 0xc000636b40, 0xc0006f2628, 0xc0000a6668, 0x119a720)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/hashicorp/terraform-plugin-sdk@v1.16.0/helper/schema/provider.go:294 +0x99
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000a62a0, 0x1501640, 0xc000574630, 0xc000572000, 0xc0000a62a0, 0xc000574630, 0xc0000b2ba0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/hashicorp/terraform-plugin-sdk@v1.16.0/internal/helper/plugin/grpc_provider.go:885 +0x8ab
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x12ebca0, 0xc0000a62a0, 0x1501640, 0xc000574630, 0xc0000a4480, 0x0, 0x1501640, 0xc000574630, 0xc000054e00, 0x35c)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      github.com/hashicorp/terraform-plugin-sdk@v1.16.0/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d6380, 0x150ce60, 0xc000601c80, 0xc0005a0000, 0xc000182b40, 0x1bf5540, 0x0, 0x0, 0x0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      google.golang.org/grpc@v1.30.0/server.go:1171 +0x522
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: google.golang.org/grpc.(*Server).handleStream(0xc0000d6380, 0x150ce60, 0xc000601c80, 0xc0005a0000, 0x0)
2021-02-11T13:44:24.442+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      google.golang.org/grpc@v1.30.0/server.go:1494 +0xcc5
2021-02-11T13:44:24.443+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00044e2d0, 0xc0000d6380, 0x150ce60, 0xc000601c80, 0xc0005a0000)
2021-02-11T13:44:24.443+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      google.golang.org/grpc@v1.30.0/server.go:834 +0xa5
2021-02-11T13:44:24.443+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-02-11T13:44:24.443+1100 [DEBUG] plugin.terraform-provider-cloudflare_v2.18.0:      google.golang.org/grpc@v1.30.0/server.go:832 +0x1fd
2021-02-11T13:44:24.444+1100 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/2.18.0/linux_amd64/terraform-provider-cloudflare_v2.18.0 pid=6038 error="exit status 2"
2021-02-11T13:44:24.444+1100 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021/02/11 13:44:24 [DEBUG] cloudflare_access_group.this["ba2_role"]: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021/02/11 13:44:24 [DEBUG] [aws-sdk-go] DEBUG: Request dynamodb/GetItem Details:

```

_Originally posted by @jgournet in https://github.com/cloudflare/terraform-provider-cloudflare/issues/939#issuecomment-777174088_

--- a/cloudflare/resource_cloudflare_access_group.go
+++ b/cloudflare/resource_cloudflare_access_group.go
@@ -552,7 +552,6 @@ func TransformAccessGroupForSchema(accessGroup []interface{}) []map[string]inter
 	azureIDs := []string{}
 	samlAttrName := ""
 	samlAttrValue := ""
-	samlID := ""
 
 	for _, group := range accessGroup {
 		for groupKey, groupValue := range group.(map[string]interface{}) {
@@ -608,7 +607,6 @@ func TransformAccessGroupForSchema(accessGroup []interface{}) []map[string]inter
 				azureIDs = append(azureIDs, azureCfg["id"].(string))
 			case "saml":
 				samlCfg := groupValue.(map[string]interface{})
-				samlID = samlCfg["identity_provider_id"].(string)
 				samlAttrName = samlCfg["attribute_name"].(string)
 				samlAttrValue = samlCfg["attribute_value"].(string)
 			case "group":
@@ -704,13 +702,12 @@ func TransformAccessGroupForSchema(accessGroup []interface{}) []map[string]inter
 		})
 	}
 
-	if samlID != "" && samlAttrName != "" && samlAttrValue != "" {
+	if samlAttrName != "" && samlAttrValue != "" {
 		data = append(data, map[string]interface{}{
 			"saml": []interface{}{
 				map[string]interface{}{
-					"identity_provider_id": samlID,
-					"attribute_name":       samlAttrName,
-					"attribute_value":      samlAttrValue,
+					"attribute_name":  samlAttrName,
+					"attribute_value": samlAttrValue,
 				}},
 		})
 	}

